### ISSUE TITLE
GPUImageBrightnessFilter affects png with transparent background

### ISSUE THREAD

**Issue by felixrosenvik** [BradLarson/GPUImage#1592]
I've been successfully using the GPUImageBrightnessFilter for months, and just recently updated to the latest version of GPUImage. The brightness filter still works well on my png images when darkening them (0 to -1), but when brightening them (0 - 1) the resulting image displayed in a UIImageView will have it's transparent background brightened aswell. I downgraded to an older version of GPUImage, and it works fine now.

Note: This does not happen in the simulator. It only occurred when running on device, in this case a iPad mini retina, iOS version 7.1.x

See below for snippet and steps to reproduce

**Comment by kocyigityunus** [comment]
@felixrosenvik  you are absolutely right. the alpha channel works on simulator as expected but not in real device. 

also 0.1.2 works as expected in simulator and real device.

**Comment by alex-dosa92** [comment]
@BradLarson Is there any solution for this? I have the same issue. On simulator everything is good but on the device the transparent pixels will be not transparent.

**Comment by rzads0** [comment]
@BradLarson @leximester  @kocyigityunus  @felixrosenvik Has anybody found a solution for this :( ? I really wish there was a solution, filters are affecting the transparent pixels still :((( I'd highly appreciate it if someone tells me a solution. Thanks guys.

**Comment by kocyigityunus** [comment]
hey @rzads0, i solved problem using v 0.1.2. 

idk if newer version work or not.

**Comment by rzads0** [comment]
@kocyigityunus  Thank you for the answer, Im using UIImageVIew and not GPUImageView, Does v 0.1.2 fix the problem of transparency in UIImageviews too? (and I have to use UIImageview Cuz no matter what I did, GPUImageview gives me so many errors.) Thank you

**Comment by rzads0** [comment]
@kocyigityunus  And I'm sorry for asking this but I tried to install V 0.1.2 and it has probs with paths so I want to install v 0.1.2 manually but having a hard time finding this version to download, Can I ask how can find this version to install it manually in my project? I totally appreciate your help.

**Comment by rzads0** [comment]
I finally managed to manually install v 0.1.2 and you're right!!! now effects don't affect transparent pixels and working great!!! Thank you so much. @kocyigityunus
